src/main/java/org/traccar/sms/SnsSmsClient.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2021 Anton Tananaev (anton@traccar.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.sms;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.sns.AmazonSNSAsync;
+import com.amazonaws.services.sns.AmazonSNSAsyncClientBuilder;
+import com.amazonaws.services.sns.model.MessageAttributeValue;
+import com.amazonaws.services.sns.model.PublishRequest;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.traccar.Context;
+import org.traccar.config.Keys;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class SnsSmsClient implements SmsManager {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SnsSmsClient.class);
+
+    private final AmazonSNSAsync snsClient;
+
+    public SnsSmsClient() {
+        if (Context.getConfig().hasKey(Keys.SMS_AWS_REGION)
+                && Context.getConfig().hasKey(Keys.SMS_AWS_ACCESS)
+                && Context.getConfig().hasKey(Keys.SMS_AWS_SECRET)) {
+            snsClient = awsSNSClient();
+        } else {
+            throw new RuntimeException("SNS Not Configured Properly. Please provide valid config.");
+        }
+    }
+
+    public AmazonSNSAsync awsSNSClient() {
```

Comment:
Please inline this.

Reply:
Can you please elaborate a little more. Perhaps an example.

Reply:
Remove this method. I already had a comment about it, but it got outdated, but not addressed.